### PR TITLE
refactor: update cxmodal search

### PR DESCRIPTION
**Pull request by JoanFM**
**Changes introduced**

Adapt cxmodal search to new code

### CODE REVIEW

File: cross-modal-search/app.py [hunk 1]
```diff
@@ -26,53 +26,56 @@ def input_index_data(num_docs=None, batch_size=8, dataset='f30k'):
...
-def main(task, num_docs, batch_size, data_set):
+@click.option('--request_size', '-s', default=16)
+@click.option('--data_set', '-d', type=click.Choice(['f30k', 'f8k'], case_sensitive=False), default='f8k')
+def main(task, num_docs, request_size, data_set):
     config()
     if task == 'index':
         f = Flow().load_config('flow-index.yml')
         with f:
-            f.index(input_fn=input_index_data(num_docs, batch_size, data_set), batch_size=batch_size)
+            f.index(input_fn=input_index_data(num_docs, request_size, data_set), batch_size=request_size)
```
Comment (cristianmtr) [inline cross-modal-search/app.py:78]:
```suggestion
            f.index(input_fn=input_index_data(num_docs, request_size, data_set), request_size=request_size)
```

Shouldn't it be like so?

Comment (JoanFM) [reply cross-modal-search/app.py:78]: in next release yes, I think no release happened yesterday

File: cross-modal-search/app.py [hunk 2]
```diff
@@ -26,53 +26,56 @@ def input_index_data(num_docs=None, batch_size=8, dataset='f30k'):
...
-    document.mime_type = 'plain/text'
+    document.mime_type = 'text/plain'
     return [document]
 
 
 def input_search_image_file(image_file_path):
     with open(image_file_path, 'rb') as fp:
         image_buffer = fp.read()
-    document = jina_pb2.Document()
+    document = Document()
```
Comment (cristianmtr) [inline cross-modal-search/app.py:61]: I think you need to do `with Document` in order to get the `content_hash`. Or call `update_content_hash` yourself. It doesn't matter for this example, but would be good that people see it when they use the example.

Comment (JoanFM) [reply cross-modal-search/app.py:61]: Okey, will use ctxt management

File: cross-modal-search/app.py [hunk 3]
```diff
@@ -26,53 +26,56 @@ def input_index_data(num_docs=None, batch_size=8, dataset='f30k'):
...
+                document.mime_type = 'text/plain'
             yield document
 
         if num_docs and (i + 1) * batch_size >= num_docs:
             break
 
 
 def input_search_text_data(text):
-    document = jina_pb2.Document()
+    document = Document()
```
Comment (cristianmtr) [inline cross-modal-search/app.py:51]: same

**Review by cristianmtr — DISMISSED** [review]
some minor things about `with Docu`

File: cross-modal-search/README.md [hunk 4]
```diff
@@ -133,33 +108,45 @@ make sure that your data folder has:
...
 ```bash
-docker build -f Dockerfile -t jinaai/hub.executors.encoders.image.vse .
-```
+pip install kaggle
+kaggle datasets download hsankesara/flickr-image-dataset
+unzip flickr-image-dataset.zip
+rm flickr-image-dataset.zip
+``` 
+
+Then we also need `captions` data, to get this:
```
Comment (cristianmtr) [inline cross-modal-search/README.md:127]:
```suggestion
Then we also need `captions` data. To get this:
```

**Review by cristianmtr — APPROVED** [review]

**Pull request merged**